src/plasma/plasma_manager.c

```diff
@@ -357,9 +357,10 @@ void remove_wait_request(plasma_manager_state *manager_state,
 void return_from_wait(plasma_manager_state *manager_state,
                       wait_request *wait_req) {
   /* Send the reply to the client. */
-  CHECK(plasma_send_WaitReply(wait_req->client_conn->fd, manager_state->builder,
-                              wait_req->object_requests,
-                              wait_req->num_object_requests) >= 0);
+  warn_if_failed(plasma_send_WaitReply(
+                     wait_req->client_conn->fd, manager_state->builder,
+                     wait_req->object_requests, wait_req->num_object_requests),
+                 wait_req->client_conn->fd);
```

Comment:
a general comment for all socket i/o handling in this PR. It might be better to specially handle errno == EPIPE and log a more useful warning message (e.g. remote client hung up). Later on, we may want to add more differentiation between various failure cases. So how about adding a handle_errno(errno) helper function here, initially with two cases: EPIPE and other, possibly even a single case borrowing errno msg logging code from the CHECK #define.